scripts/configure-venue-access.ts

```diff
@@ -95,30 +94,16 @@ initFirebaseAdminApp(projectId)(async () => {
       console.log(
         `Setting venues/${venueId}/access/${method} to {emails: ${emails}}...`
       );
-      await admin
-        .firestore()
-        .doc(`venues/${venueId}/access/${method}`)
-        .set({
-          emails: mergeStringArrays(
-            emails,
-            (access as VenueAccessEmails).emails
-          ),
-        });
+      await accessDocRef.set({
+        emails: mergeStringArrays(emails, (access as VenueAccessEmails).emails),
+      });
       break;
 
     case VenueAccessMode.Codes:
-      fs.readFileSync(accessDetail, "utf-8")
-        .split(/\r?\n/)
-        .forEach((line) => {
-          emails.push(line.trim());
-        });
       console.log(`Setting venues/${venueId}/access/${method}...`);
-      await admin
-        .firestore()
-        .doc(`venues/${venueId}/access/${method}`)
-        .set({
-          codes: (access as VenueAccessCodes).codes,
-        });
+      await accessDocRef.set({
+        codes: (access as VenueAccessCodes).codes,
+      });
```

Comment:
This seems incorrect. The code as it is left now will basically set the `codes` to the existing `codes`, aka: do nothing.

I believe the removed `fs.readFileSync` code was probably correct, but instead of pushing the lines into `email` it should have got those lines aas `codes`, then used those to update the cloud document.

---

Instead of using `.set` and merging the codes in the script, we can use something like the following:

- https://firebase.google.com/docs/firestore/manage-data/add-data#set_a_document
  - `{ merge: true }`
- https://firebase.google.com/docs/firestore/manage-data/add-data#update_elements_in_an_array
  - `admin.firestore.FieldValue.arrayUnion`